### PR TITLE
[CBR-391] launcher: provide default workingDir

### PR DESCRIPTION
**Pull request by rvl**
## Description

After #3553, this will help for the cases where launcher is not run through Daedalus, e.g. by demo.sh. 

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CBR-391

## Type of change
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## QA Steps
1. Run `cardano-launcher --config tools/src/launcher/launcher-config.yaml` and verify that it doesn't fail on `workingDir`.
2. Run `scripts/launch/demo.sh` and check that it successfully starts.

### CODE REVIEW

File: tools/src/launcher/launcher-config.yaml [hunk 1]
```diff
@@ -56,6 +56,9 @@ nodeLogConfig: log-configs/connect-to-cluster.yaml
 # File where wallet stdout/err will be redirected.
 # walletLogPath: /tmp/wallet-log
 
+# Directory to change into at startup
+# workingDir: ./
+
```
Comment (erikd) [inline tools/src/launcher/launcher-config.yaml]: What directory is that going to be when running something like `demo.sh`?

**Comment by jmitchell** [comment]
Hmm, is `x509ToolPath` required now too?

```
$ ./result/bin/cardano-launcher --config tools/src/launcher/launcher-config.yaml 
cardano-launcher: key "x509ToolPath" not present
CallStack (from HasCallStack):
  error, called at src/Universum/Debug.hs:60:11 in universum-1.2.0-ACeWmK6cPm3Ks7BVo96xtz:Universum.Debug
  error, called at src/launcher/Main.hs:207:15 in main:Main

$ git rev-parse HEAD
**02a9c6bb90f542859943f0219acebe75db44e5f0**
```

**Comment by rvl** [comment]
Yes it is. The line with `x509ToolPath` in `launcher-config.yaml` is commented out. I think because it's hard to provide a good default value. Is that right @KtorZ ?

**Comment by KtorZ** [comment]
It is required for quite a while now. The main reason _why_ is rather silly to be honest. The x509 generator is actually a binary of the `cardano-sl-tools` which means that, it can only be used as a binary and cannot expose any API (only one `library` per haskell package). Thus, even from haskell, we call the executable from the code, passing it some adequate arguments.

Ideally, we would make it its own package and have the core functions exposed as a `library` alongside a CLI as a `binary` as it's currently done. 

---

Now if the question is more like "why is this part of the launcher at all"? That's because we need to do certificates generation at some point. Doing it at install time is wrong for certificates will be the same for all users on a multi-user machine. We may also run into access permission issues if the user running the installer (typically a root-user) isn't the same actually using the program in the end.

For these reasons, it was decided to have the certificate generation done within the launcher, on the first boot.

**Comment by erikd** [comment]
Going to merge this, because the PR I'm about to raise depends on it.

**Pull request merged**